### ISSUE TITLE
[Enhancment Req'] a Warning before activating ''auto deletion after....''

### ISSUE THREAD

**Issue by zxzzz8** [oxen-io/session-desktop#2588]
just like it warn ''are you sure you want to..'' when deleting a contact or a chat history, 
add a warning before activating ''Disappearing Messages''....

btw id like to ask, does it deletes all messages in the chat history that are older then?,
or does it will delete later only messages that were created after setting that up?
coz it was very scary having the 'older then 5 seconds' right next to the OFF option, when there is no warning check,
considering that might delete my whole chat history due to a missclick.

but id like to have that warning regardless of how it works.... anything that can deletes stuff should have a warning in my opinion.

**Comment by zxzzz8** [comment]
ok, so after testing it with a friend a bit, i understand better how this works....

on that regard, i find the whole ''Disappearing Messages'' feature to be extremely troublesome on a few levels....

-

so for a start, there is no way to prevent a message from being auto deleted before the timer runs out, for example, if party A just enabled this state while party B just sent a message....
which is why i find this absurd that this doesn't apply only to your own messages,
after all you can't delete the other person's messages on their side, but the way this feature is implemented its practically a way to do exactly that....

so first i think the ''Disappearing Messages'' should only apply to your own messages when activated.

second, i think that as long as the timer didn't run out, you should be able to cancel the deletion for any specific message, at least for your own messages, in case this mode was enabled by mistake....

only after these 2 things are implemented, its ok to have no warning before turning this mode on....

-

that aside, if it was up to me personally, i wish i could disable this mode entirely, meaning that the other person cannot set this mode on, and send me temporary messages at all to begin with, even if this mode only apply to their messages, at least when sending messages to me directly.

maybe ill open other issues later on for these feature requests, but for now i guess ill wait the reaction here, as i don't want to spam here with too many feature requests etc' at once, although tbh i do have so many other feature requests i have for suggesting.

i just joined the app the other day with a friend, as i was looking for an alternative communications app for a while now, and i really hope this here is it, and so i have my hopes up for this app ^ ^ .

**Comment by KeeJef** [comment]
We are reworking disappearing messages right now, but we will be taking a similar approach to what we are now, disappearing messages when enabled are enabled for all chat participants, this keeps everyone on a level playing field and ensures when messages are deleted they are deleted from both users devices, ensuring a log cannot be recovered from either device. 

Regarding stopping automatic deletion before a message timer has expired i think this could be cool, please reopen another issue with a feature request for that, otherwise closing this issue for now
